lib/internal/Magento/Framework/View/Asset/Repository.php

```diff
@@ -146,7 +146,7 @@ public function updateDesignParams(array &$params)
         }
 
         if ($theme) {
-            $params['themeModel'] = $this->getThemeProvider()->getThemeByFullPath($area . '/' . $theme);
+            $params['themeModel'] = $this->getThemeProvider()->getThemeById($theme);
```

Comment:
Please rewrite method in such a way that if we go through line 141 `getThemeById` call is used and `getThemeByFullPath` otherwise.

Also, such case with emulated cross-area needs to be covered with integration test.